frontend/public/components/catalog-items.jsx

```diff
@@ -1,48 +1,204 @@
 import * as PropTypes from 'prop-types';
+import {Link} from 'react-router-dom';
 
 import {CatalogTileView} from 'patternfly-react-extensions/dist/esm/components/CatalogTileView';
 import {CatalogTile} from 'patternfly-react-extensions/dist/esm/components/CatalogTile';
+import {VerticalTabs} from 'patternfly-react-extensions/dist/esm/components/VerticalTabs';
+import {FilterSidePanel} from 'patternfly-react-extensions/dist/esm/components/FilterSidePanel';
 import {normalizeIconClass} from './catalog-item-icon';
-import {Link} from 'react-router-dom';
+import {categorizeItems} from './utils/categorize-catalog-items';
 
-const CatalogItem = ({item}) => {
-  const catalogTile = <CatalogTile
-    id={item.obj.metadata.uid}
-    key={item.obj.metadata.uid}
-    title={item.tileName}
-    iconImg={item.tileImgUrl}
-    iconClass={item.tileIconClass ? `icon ${normalizeIconClass(item.tileIconClass)}` : null}
-    vendor={item.tileProvider ? `Provided by ${item.tileProvider}` : null}
-    description={item.tileDescription}
-  />;
-  return item.href ? <Link className="co-catalog-item-tile" to={item.href}>{catalogTile}</Link> : catalogTile;
-};
+export class CatalogTileViewPage extends React.Component {
+  constructor(props) {
+    super(props);
 
-CatalogItem.displayName = 'CatalogItem';
-
-CatalogItem.propTypes = {
-  item: PropTypes.shape({
-    obj: PropTypes.object.isRequired,
-    kind: PropTypes.string.isRequired,
-    tileName: PropTypes.string.isRequired,
-    tileIconClass: PropTypes.string,
-    tileImgUrl: PropTypes.string,
-    tileDescription: PropTypes.string.isRequired,
-    tileProvider: PropTypes.string,
-  }).isRequired
-};
+    this.state = {
+      showAllItemsForCategory: null,
+      activeTabs: ['all'], // array of tabs [main category, sub-category]
+    };
+
+    if (props.items) {
+      this.allCategories = categorizeItems(props.items);
+      _.assign(this.state, {
+        currentCategories: this.allCategories,
+        numItems: _.first(this.allCategories).numItems,
+      });
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {items} = this.props;
+
+    if (items !== nextProps.items) {
+      this.allCategories = categorizeItems(items);
+      this.setState({
+        currentCategories: this.allCategories,
+        numItems: _.first(this.allCategories).numItems
+      });
+    }
+  }
+
+  isAllTabActive() {
+    const { activeTabs } = this.state;
+    return activeTabs[0] === 'all';
+  }
+
+  activeTabIsSubCategory(subCategories) {
+    const { activeTabs } = this.state;
+    return activeTabs.length !== 2 ? false : _.find(subCategories, ['id', activeTabs[1]]) !== undefined;
+  }
+
+  isActiveTab(categoryId) {
+    const { activeTabs } = this.state;
+    return activeTabs.length === 2 ? categoryId === activeTabs[1] : categoryId === activeTabs[0];
+  }
+
+  doesHaveActiveDescendants(categoryId) {
+    const { activeTabs } = this.state;
+    return activeTabs[0] === categoryId;
```

Comment:
This is fine if limiting to 2 levels. You could check if any of the activeTabs have the categoryId to support more levels. Better to keep just one active tab Id, and determine if a parent has an active descendant based on that.